vllm/v1/core/sched/scheduler.py

```diff
@@ -779,8 +779,15 @@ def update_from_output(
                 # NOTE: structured_output_request
                 # should not be None if use_structured_output, we have
                 # check above, so safe to ignore type warning
-                request.structured_output_request.grammar.accept_tokens(  # type: ignore[union-attr]
-                    req_id, new_token_ids)
+                if not request.structured_output_request.grammar.accept_tokens(  # type: ignore[union-attr]
+                        req_id, new_token_ids):
```

Comment:
Suggest using a variable here to make things a bit clearer:
```suggestion
                accepted = request.structured_output_request.grammar.accept_tokens(  # type: ignore[union-attr]
                        req_id, new_token_ids)
                if not accepted:
```